drift-transport-netty/src/main/java/io/airlift/drift/transport/netty/codec/HeaderTransport.java

```diff
@@ -240,4 +244,38 @@ private static int readVariableLengthInt(ByteBuf messageHeader)
 
         return result;
     }
+
+    public static Optional<FrameInfo> tryDecodeFrameInfo(ByteBuf input)
+    {
+        ByteBuf buffer = input.retainedDuplicate();
+        try {
+            if (buffer.readableBytes() < FRAME_HEADER_SIZE) {
+                return Optional.empty();
+            }
+            // skip magic
+            buffer.readShort();
+            short flags = buffer.readShort();
+            boolean outOfOrderResponse = isOutOrOrderResponseSupported(flags);
+            int sequenceId = buffer.readInt();
+            int headerSize = buffer.readShort() << 2;
+
+            if (buffer.readableBytes() < Byte.BYTES) {
+                return Optional.empty();
+            }
+            byte protocolId = buffer.getByte(buffer.readerIndex());
+            Protocol protocol = Protocol.getProtocolByHeaderTransportId(protocolId);
+
+            if (buffer.readableBytes() < headerSize) {
```

Comment:
I think this check needs to be above the `buffer.getByte()` for the protocol ID, since that is included in the header size (based on my reading of `decodeFrame()`)